examples/ucnf-kiknos/scripts/dump_clusters_state.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+# Topology information
+CLUSTER1=${CLUSTER1:-kiknos-demo-1}
+CLUSTER2=${CLUSTER2:-kiknos-demo-2}
+
+
+pushd "$(dirname "$0")/../../../"
+
+print_usage() {
+  echo "$(basename "$0") - Deploy NSM Kiknos topology. All properties can also be provided through env variables
+
+NOTE: The defaults will change to the env values for the ones set.
+
+Usage: $(basename "$0") [options...]
+Options:
+  --cluster1            Name of Kind cluster one - Represents the client network            env var: CLUSTER1         - (Default: $CLUSTER1)
+  --cluster2            Name of Kind cluster two - Represents the VPN Gateway               env var: CLUSTER2         - (Default: $CLUSTER2)
+" >&2
+
+}
+
+for i in "$@"; do
+  case $i in
+  --cluster1=*)
+    CLUSTER1="${i#*=}"
+    ;;
+  --cluster2=*)
+    CLUSTER2="${i#*=}"
+    ;;
+  -h | --help)
+    print_usage
+    exit 0
+    ;;
+  *)
+    print_usage
+    exit 1
+    ;;
+  esac
+done
+
+function dump_state_of_namespace() {
+    local cluster=$1; shift
+    local namespace=$1; shift
+    echo ""
+    echo "##################################### Dumping state of $namespace for $cluster #########################################"
+    kubectl get pods --context "$cluster" -n "$namespace"
+}
+
+function dump_addresses_of_endpoints() {
+  local cluster=$1; shift
+  echo
+  echo "##################################### Dumping NSE interfaces for $cluster #########################################"
+  for NSE in $(kubectl --context "$cluster" get pods -l networkservicemesh.io/impl=vpn-endpoint -o=name); do
+    echo "--------------------------------------- NSE $NSE interfaces --------------------------------------------------"
+    echo "-------------------------------------------Addresses----------------------------------------------------------"
+    kubectl --context "$cluster" exec -it "$NSE" -- vppctl show interf addr
+    echo "-------------------------------------------Interfaces---------------------------------------------------------"
+    kubectl --context "$cluster" exec -it "$NSE" -- vppctl show interf
+    echo "---------------------------------------------Mode-------------------------------------------------------------"
+    kubectl --context "$cluster" exec -it "$NSE" -- vppctl sh mode
+    echo "-------------------------------------------VxLan Tunnel-------------------------------------------------------"
+    kubectl --context "$cluster" exec -it "$NSE" -- vppctl sh vxlan tunnel
+    echo "----------------------------------------------Fib-------------------------------------------------------------"
+    kubectl --context "$cluster" exec -it "$NSE" -- vppctl sh ip fib
```

Comment:
Command notes that I have:
```
vppctl commands:
----------------

  sh int
  sh int addr
  sh mode
  sh ipip tunnel
    - shows the src and dest of the IPIP tunnel
  sh ipsec sa
    - summary view of the IPSEC security associations (SA)
      NOTE: for a single connection there's 1 per direction (input/output)
  sh ipsec protect
    - shows the SA direction mapping and interface association
  sh ipsec sa 0
    - "detail" view of SA index 0 (index = 0 seems to be output, index = 1 input)
  sh ipsec sa 1
    - "detail" view of SA index 1 (index = 0 seems to be output, index = 1 input)

  NOTE: for kiknos NSEs the IKEv2 commands don't show anything 
  - sh ikev2 sa
  - sh ikev2 profile

  Miscellaneous commands that possibly show something interesting:
  - sh crypto engines
  - sh crypto handlers
  - sh log
  
```

Reply:
I have updated the tech dump with the mentioned commands.